app/models/hyrax/batch_ingest/ability.rb

```diff
@@ -12,38 +12,17 @@ def batch_abilities
-          can [:new], Hyrax::BatchIngest::Batch if Hyrax::Collections::PermissionsService.source_ids_for_deposit(ability: self, source_type: 'admin_set').present?
-
-          # user who can deposit into an admin set can create batch for the admin set
-          can [:create], Hyrax::BatchIngest::Batch do |batch|
-            admin_set = batch.admin_set_id.present? && AdminSet.exists?(batch.admin_set_id) ? AdminSet.find(batch.admin_set_id) : nil
-            admin_set && (can? :deposit, admin_set)
-          end
-          # can [:create], Hyrax::BatchIngest::Batch do |batch|
-          #   Hyrax::Collections::PermissionsService.can_deposit_in_collection?(ability: self, collection_id: batch.admin_set_id)
-          # end
-
-          # user who can view at least one admin set can access the batch index
-          can [:index], Hyrax::BatchIngest::Batch if can? :view_admin_show_any, AdminSet
-          # can [:index], Hyrax::BatchIngest::Batch if Hyrax::Collections::PermissionsService.can_view_admin_show_for_any_admin_set?(ability: self)
-
-          # manager of a admin set (i.e. who can update the admin set) can show/cancel any batch for that admin set
-          can [:show, :destroy], Hyrax::BatchIngest::Batch do |batch|
-            admin_set = batch.admin_set_id.present? && AdminSet.exists?(batch.admin_set_id) ? AdminSet.find(batch.admin_set_id) : nil
-            admin_set && (can? [:edit, :update, :destroy, :view_admin_show], admin_set)
-          end
-          # can [:show, :destroy], Hyrax::BatchIngest::Batch do |batch|
-          #   Hyrax::Collections::PermissionsService.manage_access_to_collection?(collection_id: batch.admin_set_id, ability: self)
-          # end
-
-          # depositor of a batch can only show/cancel the batch created by himself
-          can [:show, :destroy], Hyrax::BatchIngest::Batch do |batch|
-            current_user.email == batch.submitter_email
+          deposit_admin_sets = Hyrax::Collections::PermissionsService.source_ids_for_deposit(ability: self, source_type: 'admin_set')
+          if deposit_admin_sets.present?
+            # user who can deposit into at least one admin set can initiate a new batch
+            can :new, Hyrax::BatchIngest::Batch
+            # user who can deposit into an admin set can create batch for the admin set
+            can :create, Hyrax::BatchIngest::Batch, admin_set_id: deposit_admin_sets
+            # depositor of a batch can only show their batch
+            can :read, Hyrax::BatchIngest::Batch, submitter_email: current_user.email
+            # manager of a admin set (i.e. who can update the admin set) can show any batch for that admin set
+            can :read, Hyrax::BatchIngest::Batch, admin_set_id: Hyrax::Collections::PermissionsService.source_ids_for_manage(ability: self, source_type: 'admin_set')
           end
-          # can [:show, :destroy], Hyrax::BatchIngest::Batch do |batch|
```

Comment:
I had to go back to the Cancancan documentation since I was getting errors due to `accessible_by` being incompatible with block style abilities.  Cancancan depends on `accessible_by` to load the instances for the `:index` action and I didn't really want to have to add an exception for it and do the filtering ourselves in the controller.  I've never set the abilities up like this before to avoid the blocks and I'm still a little concerned about when this gets evaluated and if it is too static.  I guess we can deal with that if/when we receive a user complaint.